snuba/datasets/replays_processor.py

```diff
@@ -27,146 +27,121 @@
-        self, message: Mapping[Any, Any]
-    ) -> Optional[Tuple[EventDict, RetentionDays]]:
-
-        event = message
-        data = event["data"]
-
-        try:
-            # We are purposely using a naive datetime here to work with the
-            # rest of the codebase. We can be confident that clients are only
-            # sending UTC dates.
-            retention_days = enforce_retention(
-                message["retention_days"], datetime.utcfromtimestamp(data["timestamp"])
-            )
-        except EventTooOld:
-            return None
-
-        return event, retention_days
-
-    def _process_base_event_values(
-        self, processed: MutableMapping[str, Any], event_dict: EventDict
-    ) -> MutableMapping[str, Any]:
-
-        processed["replay_id"] = str(uuid.UUID(event_dict["replay_id"]))
-        processed["project_id"] = event_dict["project_id"]
-
-        processed["sequence_id"] = event_dict["sequence_id"]
-        processed["trace_ids"] = event_dict["trace_ids"]
-
+    def _add_user_field_tag(
+        self, processed: Any, replay_event: ReplayEventDict
+    ) -> None:
+        if "user" not in replay_event:
+            return
+        for field in USER_FIELDS_PRECEDENCE:
+            if field in replay_event["user"]:
+                processed["user"] = replay_event["user"][field]
+                return
+
+    def _get_url(self, replay_event: ReplayEventDict) -> Optional[str]:
+        if "request" in replay_event:
+            if "url" in replay_event["request"]:
+                return cast(str, replay_event["request"]["url"])
+        return None
+
+    def _process_base_replay_event_values(
+        self, processed: MutableMapping[str, Any], replay_event: ReplayEventDict
+    ) -> None:
+        processed["replay_id"] = str(uuid.UUID(replay_event["replay_id"]))
+        processed["sequence_id"] = replay_event["sequence_id"]
+        processed["trace_ids"] = replay_event.get("trace_ids")
         processed["timestamp"] = self.__extract_timestamp(
-            event_dict["data"]["timestamp"],
+            replay_event["timestamp"],
         )
-
-        processed["platform"] = _unicodify(event_dict["platform"])
-        return processed
+        processed["release"] = replay_event.get("release")
+        processed["environment"] = replay_event.get("environment")
+        processed["dist"] = replay_event.get("dist")
+        processed["url"] = self._get_url(replay_event)
+        processed["platform"] = _unicodify(replay_event["platform"])
 
     def _process_tags(
-        self,
-        processed: MutableMapping[str, Any],
-        event_dict: EventDict,
+        self, processed: MutableMapping[str, Any], replay_event: ReplayEventDict
     ) -> None:
-
-        tags: Mapping[str, Any] = _as_dict_safe(event_dict["data"].get("tags", None))
+        tags: MutableMapping[str, Any] = _as_dict_safe(replay_event.get("tags", None))
+        processed["title"] = tags.get("transaction")
```

Comment:
`title` is not nullable.

Reply:
👍🏼 let me know if it's preferred to write empty string, or we can also make it nullable instead. i saw this: https://stackoverflow.com/questions/63057886/clickhouse-string-field-disk-usage-null-vs-empty